cmd/agent/executor_agent_service.go

```diff
@@ -0,0 +1,229 @@
+	if exec == nil {
+		return fmt.Errorf("Cannot add a nil Executor")
+	}
+
+	e.executors[exec.GetAccountName()] = exec
+
+	return nil
+}
+
+// createExecutors initializes CloudExecutors for all configured cloud provider accounts.
+//
+// Returns:
+//   - error: An error if any executor initialization fails.
+func (e *ExecutorAgentService) createExecutors() error {
+	accounts, err := e.readCloudProviderAccounts()
+	if err != nil {
+		return err
+	}
+
+	// Generating a CloudExecutor by account. The creation of the CloudExecutor depends on the Cloud Provider
+	for _, account := range accounts {
+		switch account.Provider {
+		case inventory.AWSProvider: // AWS
+			e.logger.Info("Creating Executor for AWS account", zap.String("account_name", account.Name))
+			exec := cexec.NewAWSExecutor(
+				inventory.NewAccount("", account.Name, account.Provider, account.User, account.Key),
+				e.actionsChannel,
+				logger,
+			)
+			err := e.AddExecutor(exec)
+			if err != nil {
+				e.logger.Error("Cannot create an AWSEexecutor for account", zap.String("account_name", account.Name), zap.Error(err))
+				return err
+			}
+
+		case inventory.GCPProvider: // GCP
+			e.logger.Warn("Failed to create Executor for GCP account",
+				zap.String("account", account.Name),
+				zap.String("reason", "not implemented"),
+			)
+
+		case inventory.AzureProvider: // Azure
+			e.logger.Warn("Failed to create Executor for Azure account",
+				zap.String("account", account.Name),
+				zap.String("reason", "not implemented"),
+			)
+
+		}
+	}
+	return nil
+}
+
+// GetExecutor retrieves the CloudExecutor associated with a given account name.
+//
+// Parameters:
+// - accountName: The name of the account for which the executor is requested.
+//
+// Returns:
+// - cexec.CloudExecutor: The executor for the specified account.
+// - error: An error if no executor is found for the given account.
+func (e *ExecutorAgentService) GetExecutor(accountName string) *cexec.CloudExecutor {
+	exec, ok := e.executors[accountName]
+	if !ok {
+		return nil
+	}
+	return &exec
+}
+
+func (e *ExecutorAgentService) Start() error {
+	e.logger.Debug("Starting ExecutorAgentService")
+	var actionStatus string
+
+	for action := range e.actionsChannel {
+		e.logger.Debug("New action arrived to ExecutorAgentService",
+			zap.Any("action", action.GetActionOperation()),
+			zap.Any("target", action.GetTarget()),
+		)
+
+		description := "ScheduledAction(" + action.GetID() + ")"
+
+		// Initialize event tracker
+		tracker := e.eventService.StartTracking(&events.EventOptions{
+			Action:       inventory.ClusterPowerOnAction,
+			Description:  &description,
+			ResourceID:   action.GetTarget().ClusterID,
+			ResourceType: inventory.ClusterResourceType,
+			Result:       events.ResultPending,
+			Severity:     events.SeverityInfo,
+			TriggeredBy:  "ClusterIQ Agent",
+		})
+
+		target := action.GetTarget()
+		cexec := *(e.GetExecutor(target.GetAccountName()))
```

Comment:
I guess would it be better to avoid package collides

Reply:
replaced `action` var name by `newAction` to avoid conflicts

Reply:
replaced `action` var name by `newAction` to avoid conflicts